src/js/components/Notification/Notification.js

```diff
@@ -63,9 +64,13 @@ const Notification = ({
 }) => {
   const autoClose =
     toast && toast?.autoClose === undefined ? true : toast.autoClose;
+  const size = useContext(ResponsiveContext);
   const theme = useContext(ThemeContext) || defaultProps.theme;
   const [visible, setVisible] = useState(true);
-  const position = useMemo(() => (toast && toast?.position) || 'top', [toast]);
+
+  const position = useMemo(() => 
+    (size === 'small') ? 'top' : (toast && toast?.position) || 'top', 
+    [size, toast]);
```

Comment:
A use case that will be tricky with this logic, is what would happen when the user defines an extended t-shirt sizing range (custom breakpoints), such as "xsmall"? 
https://storybook.grommet.io/?path=/story/utilities-responsivecontext-custom-breakpoints--custom-breakpoints

Reply:
Good comment ^^ was thinking the same. (noticed we implemented this logic in NameValueList and will likely have to think of a better approach/mirror what we land on here)